webview/src/plots/components/customPlots/CustomPlot.tsx

```diff
@@ -13,26 +18,38 @@ interface CustomPlotProps {
   id: string
 }
 
+const createCustomPlotSpec = (
+  plot: CustomPlotData | undefined,
+  colors: ColorScale | undefined
+) => {
+  if (!plot) {
+    return {}
+  }
+
+  if (isCheckpointPlot(plot)) {
+    return createCheckpointSpec(plot.yTitle, plot.param, colors)
+  }
+  return createMetricVsParamSpec(plot.yTitle, plot.param)
```

Comment:
[Q] Does this still need to be in the webview? Can it be moved back to the extension?

The reason that I ask is that we want to limit the number of places `isCheckpointPlot`/`isCheckpointValue` get called. Having the logic spread out through different parts of the codebase will lead to bugs. 

Note: Not everyone agrees but type-checking can be considered a code smell. I think it is ok if it is contained to a single part of the codebase/a few files but the more it spreads the more likely it is to cause bugs.

Reply:
> [Q] Does this still need to be in the webview? Can it be moved back to the extension?

It looks like it can be sent via the extension... But I'm unaware of why it was in the webview in the first place 🤔